ui/ducks/send/send.js

```diff
@@ -632,36 +666,8 @@ export const initializeSendState = createAsyncThunk(
     // We have to keep the gas slice in sync with the send slice state
     // so that it'll be initialized correctly if the gas modal is opened.
     await thunkApi.dispatch(setCustomGasLimit(gasLimit));
-    // We must determine the balance of the asset that the transaction will be
-    // sending. This is done by referencing the native balance on the account
-    // for native assets, and calling the balanceOf method on the ERC20
-    // contract for token sends.
-    let { balance } = account;
-    if (asset.type === ASSET_TYPES.TOKEN) {
-      if (asset.details === null) {
-        // If we're sending a token but details have not been provided we must
-        // abort and set the send slice into invalid status.
-        throw new Error(
-          'Send slice initialized as token send without token details',
-        );
-      }
-      balance = await getERC20Balance(asset.details, fromAddress);
-    }
-
-    if (asset.type === ASSET_TYPES.COLLECTIBLE) {
-      if (asset.details === null) {
-        // If we're sending a collectible but details have not been provided we must
-        // abort and set the send slice into invalid status.
-        throw new Error(
-          'Send slice initialized as collectibles send without token details',
-        );
-      }
-      balance = '0x1';
-    }
```

Comment:
this portion is now handled in the updateSendAsset, which now is never called from outside the send flow (it was previously called before routing to the send page to set the asset to be sent)